comictaggerlib/volumeselectionwindow.py

```diff
@@ -334,6 +345,40 @@ def searchComplete(self):
             return
 
         self.cv_search_results = self.search_thread.cv_search_results
+        # filter the blacklisted publishers if setting set
+        if self.use_blackList:
+            try:
+                publisher_blacklist = set([s.strip().lower() for s in self.settings.id_publisher_blacklist.split(',')])
+                # use '' as publisher name if None
+                self.cv_search_results = list(filter(lambda d: ('' if d['publisher'] is None else str(d['publisher']['name']).lower()) not in publisher_blacklist, self.cv_search_results))
+            except:
+                print('bad data error filtering blacklist publishers')
+
+        # pre sort the data - so that we can put exact matches first afterwards
+        # compare as str incase extra chars ie. '1976?'
+        # - missing (none) values being converted to 'None' - consistant with prior behaviour in v1.2.3
+        # sort by start_year if set
+        if self.settings.sort_series_by_year:
+            try:
+                self.cv_search_results = sorted(self.cv_search_results, key = lambda i: (str(i['start_year']), str(i['count_of_issues'])), reverse=True)
+            except:
+                print('bad data error sorting results by start_year,count_of_issues')
+        else:
+            try:
+                self.cv_search_results = sorted(self.cv_search_results, key = lambda i: str(i['count_of_issues']), reverse=True)
+            except:
+                print('bad data error sorting results by count_of_issues')
```

Comment:
natsort handles different types well also operator.itemgetter makes simple keys much easier

```suggestion
        # pre sort the data, so that we can put exact matches first
        # sort by start_year if set
        if self.settings.sort_series_by_year:
            try:
                self.cv_search_results = natsorted(self.cv_search_results, key=itemgetter('start_year', 'count_of_issues'), reverse=True)
            except:
                print('error sorting results by start_year,count_of_issues')
        else:
            try:
                self.cv_search_results = natsorted(self.cv_search_results, key=itemgetter('count_of_issues'), reverse=True)
            except:
                print('error sorting results by count_of_issues')
```

Reply:
This fails sorting records containing None values
when you search Comic Vine for a series called 'Excalibur', one of the results is a bed result that just has None in the other fields

Reply:
This is true I only tested when comparing strings and None which are comparable however numeric types are not comparable with None